src/spyglass/utils/misc.py

```diff
@@ -0,0 +1,18 @@
+import uuid
+
+
+def generate_nwb_uuid(nwb_file_name: str, initial: str, len_uuid: int = 6):
+    """Generates a unique identifier related to an NWB file.
+
+    Parameters
+    ----------
+    nwb_file_name : str
+        _description_
+    initial : str
+        R if recording; A if artifact; S if sorting etc
+    len_uuid : int
+        how many digits of uuid4 to keep
+    """
+    uuid4 = uuid.uuid4()
+    nwb_uuid = nwb_file_name + "_" + initial + "_" + uuid4[:len_uuid]
+    return nwb_uuid
```

Comment:
I agree it's not ideal. The goal was to use unique IDs as the sole primary key for some of the tables (e.g. `recording_id` for `SpikeSortingRecording`, `sorting_id` for `SpikeSorting`) so that we don't run into the problem of having very long composite primary key downstream. I thought putting the raw nwb file name and the initial in front of the partial uuid would give one some sense of where the data came from and what type of analysis nwb file this is while making sure it is unique. If you have a better approach, I would be happy to hear it.

Reply:
Would the nwb_file_name, interval_list_name, etc. still be in secondary keys? If not it could be difficult to fetch the spikesorting data you need during analysis

Reply:
@samuelbray32 
Right now, in the case of `SpikeSortingRecording`, its primary key is `recording_id` (inherited from `SpikeSortingRecordingSelection`). Its secondary keys are `analysis_file_name` and `object_id`, corresponding to the output of the `populate` call. The other critical information (`nwb_file_name`, `interval_list_name` and so on) is stored in `SpikeSortingRecordingSelection` as secondary key. so you would fetch it from there e.g. `(SpikeSortingRecordingSelection & [recording_id]).fetch1('nwb_file_name')`.

I initially had these as primary key in  `SpikeSortingRecordingSelection` (as in v0 pipeline) and secondary (foreign) key in `SpikeSortingRecording` as you suggest here, but then I lose the primary key dependency between `SpikeSortingRecordingSelection` and `SpikeSortingRecording` (as you have pointed out in a different comment) and `populate` doesn't work. So my workaround has been to introduce the `insert_selection` method (that I described in yet another comment thread). 

Does this seem OK to you?